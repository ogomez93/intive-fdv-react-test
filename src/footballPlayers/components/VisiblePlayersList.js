import React from 'react';
import { connect } from 'react-redux';
import Player from './Player';
import { getVisiblePlayers } from '../selectors';

const VisiblePlayersList = ({ players }) => (
  <tbody>
    {
      !players.length ? 
        <tr id="noPlayersMatchFilters">
          <td className="border-2 border-black px-1 py-3 sm:py-6" colSpan="4">No players match those filters</td>
        </tr>
      :
        players.map(player => <Player player={player} key={player.name} />)
    }
  </tbody>
);

const mapStateToProps = (state) => {
  return {
    players: getVisiblePlayers(state)
  }
}

export default connect(
  mapStateToProps
)(VisiblePlayersList);