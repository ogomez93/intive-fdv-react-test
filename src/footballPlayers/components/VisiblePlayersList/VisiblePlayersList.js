import React from 'react';
import Player from 'footballPlayers/components/Player';

const VisiblePlayersList = ({ players }) => (
  <tbody>
    { players.map(player => <Player {...player} key={player.name} />) }
  </tbody>
);

export default VisiblePlayersList
