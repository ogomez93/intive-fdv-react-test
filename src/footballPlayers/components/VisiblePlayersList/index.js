import { connect } from 'react-redux';
import { branch, compose, renderComponent } from 'recompose';
import VisiblePlayersList from './VisiblePlayersList';
import NoPlayers from 'footballPlayers/components/NoPlayers';
import { getVisiblePlayers } from 'footballPlayers/selectors';

const mapStateToProps = state => ({
  players: getVisiblePlayers(state)
});

export default compose(
  connect(mapStateToProps),
  branch(({ players }) => players.length === 0, renderComponent(NoPlayers))
)(VisiblePlayersList);
