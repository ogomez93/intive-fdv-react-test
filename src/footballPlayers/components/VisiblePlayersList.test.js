import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import VisiblePlayersList from './VisiblePlayersList';
import {
  initialStateWithoutPlayers,
  initialStateWithTwoPlayers,
  initialStateWithTwoPlayersAndFilter,
  initialStateWithTwoPlayersAndUnexistentFilter
} from './testConstants/VisiblePlayersList.constants';

const mockStore = configureStore();

const storeWithoutPlayers = mockStore(initialStateWithoutPlayers);
const storeWithTwoPlayers = mockStore(initialStateWithTwoPlayers);
const storeWithTwoPlayersAndFilter = mockStore(initialStateWithTwoPlayersAndFilter);
const storeWithTwoPlayersAndUnexistentFilter = mockStore(initialStateWithTwoPlayersAndUnexistentFilter);

// without players

describe('Component: <VisiblePlayersList /> Without Players', () => {
  const wrapper = shallow(<VisiblePlayersList store={storeWithoutPlayers} />);
  const component = wrapper.dive();
  const componentToJson = toJson(component);
  const tr = componentToJson.children[0];

  test('renders the component', () => {
    expect(componentToJson).toMatchSnapshot();
  });

  test('no players match the filters', () => {
    expect(tr.node.rendered.rendered).toEqual('No players match those filters');
  });
});

// with 2 players

describe('Component: <VisiblePlayersList /> With 2 Players', () => {
  const wrapper = shallow(<VisiblePlayersList store={storeWithTwoPlayers} />);
  const component = wrapper.dive();
  const componentToJson = toJson(component);

  test('renders the component', () => {
    expect(componentToJson).toMatchSnapshot();
  });

  test('renders 2 players', () => {
    const players = componentToJson.children.filter((el) => el.type === 'Player');
    expect(players.length).toEqual(2);
  });
});

// with 2 players and name filter

describe('Component: <VisiblePlayersList /> With 2 Players and name filter', () => {
  const wrapper = shallow(<VisiblePlayersList store={storeWithTwoPlayersAndFilter} />);
  const component = wrapper.dive();
  const componentToJson = toJson(component);

  test('renders the component', () => {
    expect(componentToJson).toMatchSnapshot();
  });

  test('renders 1 player', () => {
    const players = componentToJson.children.filter((el) => el.type === 'Player');
    expect(players.length).toEqual(1);
  });
});

// with 2 players and unexistent name filter

describe('Component: <VisiblePlayersList /> With 2 Players and unexistent name filter', () => {
  const wrapper = shallow(<VisiblePlayersList store={storeWithTwoPlayersAndUnexistentFilter} />);
  const component = wrapper.dive();
  const componentToJson = toJson(component);

  test('renders the component', () => {
    expect(componentToJson).toMatchSnapshot();
  });

  test('renders 0 players', () => {
    const players = componentToJson.children.filter((el) => el.type === 'Player');
    expect(players.length).toEqual(0);
  });

  test('no players match the filters', () => {
    const td = componentToJson.children[0].children[0];
    expect(td.children[0]).toEqual('No players match those filters');
  });
});
