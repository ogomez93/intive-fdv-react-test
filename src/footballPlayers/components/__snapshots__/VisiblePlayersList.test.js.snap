// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <VisiblePlayersList /> With 2 Players and name filter renders the component 1`] = `
<tbody>
  <Player
    key="test name 1"
    player={
      Object {
        "dateOfBirth": "2000-01-01",
        "name": "test name 1",
        "nationality": "test nationality 1",
        "position": "Central Midfield",
      }
    }
  />
</tbody>
`;

exports[`Component: <VisiblePlayersList /> With 2 Players and unexistent name filter renders the component 1`] = `
<tbody>
  <tr
    id="noPlayersMatchFilters"
  >
    <td
      className="border-2 border-black px-1 py-3 sm:py-6"
      colSpan="4"
    >
      No players match those filters
    </td>
  </tr>
</tbody>
`;

exports[`Component: <VisiblePlayersList /> With 2 Players renders the component 1`] = `
<tbody>
  <Player
    key="test name 1"
    player={
      Object {
        "dateOfBirth": "2000-01-01",
        "name": "test name 1",
        "nationality": "test nationality 1",
        "position": "Central Midfield",
      }
    }
  />
  <Player
    key="test name 2"
    player={
      Object {
        "dateOfBirth": "2000-01-01",
        "name": "test name 2",
        "nationality": "test nationality 2",
        "position": "Centre-Back",
      }
    }
  />
</tbody>
`;

exports[`Component: <VisiblePlayersList /> Without Players renders the component 1`] = `
<tbody>
  <tr
    id="noPlayersMatchFilters"
  >
    <td
      className="border-2 border-black px-1 py-3 sm:py-6"
      colSpan="4"
    >
      No players match those filters
    </td>
  </tr>
</tbody>
`;
